README_docker.md

```diff
@@ -0,0 +1,164 @@
+
+	docker-compose up --build
+
+Finally, the image is ready for DKG. Just start it and let it run in background with 
+
+	docker-compose up -d
+
+(one can also set a restart policy in the docker-compose.yml)
+
+## DKG
+
+If you did the setup above, you have a container running, loaded with your keys. It still misses two things:
+
+1. the group.toml file corresponding to other participants. For this, you have to exchange keys manually, e.g., via email.
+2. running the DKG protocol to bootstrap drand.
+
+Fortunately, with our docker-compose volumes, it's now very easy to add things into the running container. Just add your `group.toml` into the root of the `data` folder (__NOT__ in the `data/groups/` folder; this one is manually managed by drand, don't touch it).
+
+Then, open a CLI into your running docker.
+
+First find its id on the host:
+
+	$ docker ps
+	697e4766f8b2        drand_drand             "drand --verbose 2 s…"   11 minutes ago      Up 9 minutes
+
+The id of the container is `697e4766f8b2`. Enter it by running:
+
+	docker exec -it 697e4766f8b2 /bin/sh
+
+Then, you're inside the container; tell drand to run the DKG like so:
+
+	drand share /root/.drand/group.toml
+
+__Notice the full path__ `/root/.drand/group.toml` and not `group.toml` nor `./group.toml`
+
+## Checking the logs
+
+From outside the container, run `docker-compose logs`
+
+## Updating drand
+
+To update the docker, simply shut it down
+
+	docker-compose down
+
+and to fully rebuild it, you need to first clean the already-used layers (for some reason Docker is confused and thinks nothing has changed, and it keeps rebuilding the old version)
+
+	docker system prune -a
+
+Caution! this delete *all* your *unused* containers and networks; it's typically fine, but just be aware of it.
+
+Then rebuild and restart it
+
+	docker-compose up --build -d
+
+## Something went wrong, reset (without losing the keys)
+
+Kill the container:
+
+	docker-compose down
+
+Delete what you want to reset in `data`: typically, you absolutely want to __keep__ `data/keys`, __especially__ if you shared those keys to create a `group.toml` with other people. For instance if the DKG failed, remove `data/db` and `data/groups`. Notice that if you added the `group.toml` into the root of `data` as suggested, it should still be there (don't delete it unless you want to change the group).
+
+Then, rebuild the image from scratch:
+
+	docker system prune -a
+	docker-compose up --build -d
+
+Check that things are running with
+
+	docker-compose logs
+
+You're now back to the step "DKG" of this guide.
+
+## Docker behind reverse proxy setup
+
+Typically, the TLS part of my VPS is managed by a single reverse proxy, which then proxies multiple services running locally with docker.
+
+There is one subtletly: you need to forward _both_ GRPC (used by drand "core") and web traffic (for the web interface). To forward GRPC, you need to have nginx `1.13.10` or above, it's a fairly recent addition.
```

Comment:
That's so cool that it works now !! awesome

Reply:
Yes that's nice ;)